R/tokens_group.R

```diff
@@ -22,24 +24,26 @@
 #' # with fill
 #' tokens_group(toks, groups = factor(c(1, 1, 2, 2), levels = 1:3))
 #' tokens_group(toks, groups = factor(c(1, 1, 2, 2), levels = 1:3), fill = TRUE)
-tokens_group <- function(x, groups = docid(x), fill = FALSE) {
+tokens_group <- function(x, groups = docid(x), fill = FALSE, env = NULL) {
     UseMethod("tokens_group")
 }
 
 #' @export
-tokens_group.default <- function(x, groups = docid(x), fill = FALSE) {
+tokens_group.default <- function(x, groups = docid(x), fill = FALSE, env = NULL) {
     check_class(class(x), "tokens_group")
 }
 
 #' @export
-tokens_group.tokens <- function(x, groups = docid(x), fill = FALSE) {
-    x <- as.tokens(x)
+tokens_group.tokens_xptr <- function(x, groups = docid(x), fill = FALSE, env = NULL) {
+    
+    if (is.null(env))
+        env <- parent.frame()
     if (missing(groups)) {
         field <- NULL
         groups <- docid(x)
     } else {
         field <- deparse(substitute(groups))
-        groups <- eval(substitute(groups), get_docvars(x, user = TRUE, system = TRUE), parent.frame())
```

Comment:
I think we specified `parent.frame()` to make sure that this would work in other functions or environments, not just the base environment. Have you tested this under such conditions?

Reply:
I think these tests covers all the possible cases. 

https://github.com/quanteda/quanteda/blob/d737996e45061fddb36432a6d104177f445b14b0/tests/testthat/test-tokens_group.R#L127-L156